core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -524,6 +524,51 @@ case class PSBT(
     PSBT(globalMap, newInputMaps, outputMaps)
   }
 
+  def verifyFinalizedInput(index: Int): Boolean = {
+    val inputMap = inputMaps(index)
+    require(inputMap.isFinalized, "Input must be finalized to verify")
+
+    val wUtxoOpt = inputMap.witnessUTXOOpt
+    val utxoOpt = inputMap.nonWitnessOrUnknownUTXOOpt
+
+    val newInput = {
+      val input = transaction.inputs(index)
+      val scriptSigOpt = inputMap.finalizedScriptSigOpt
+      val scriptSig =
+        scriptSigOpt.map(_.scriptSig).getOrElse(EmptyScriptSignature)
+      TransactionInput(input.previousOutput, scriptSig, input.sequence)
+    }
+
+    val tx = transaction.updateInput(index, newInput)
+
+    wUtxoOpt match {
+      case Some(wUtxo) =>
+        inputMap.finalizedScriptWitnessOpt match {
+          case Some(scriptWit) =>
+            val wtx = {
+              val wtx = WitnessTransaction.toWitnessTx(transaction)
+              wtx.updateWitness(index, scriptWit.scriptWitness)
+            }
+            val output = wUtxo.witnessUTXO
+
+            ScriptInterpreter.verifyInputScript(wtx, index, output)
```

Comment:
Do we run into issues if there is a `p2wsh` elsewhere which has redeem script witnesses that aren't included?

Reply:
Will take a look and make a PR on master if there is

Reply:
Looks like it should be good in PSBTs